indy_node/test/request_handlers/test_context_handler.py

```diff
@@ -0,0 +1,267 @@
+    # change true to True to correct for python
+    # Sample from specification: https://w3c.github.io/vc-data-model/#base-context
+    # Actual file contents from: https://www.w3.org/2018/credentials/v1
+    validate_context(W3C_BASE_CONTEXT)
+
+
+def test_static_validation_pass_valid_transaction():
+    operation = {
+        "meta": {
+            "name": "TestContext",
+            "version": 1,
+            "type": CONTEXT_TYPE
+        },
+        "data": W3C_BASE_CONTEXT,
+        "type": "200"
+    }
+    req = Request("test", 1, operation, "sig",)
+    ch = ContextHandler(None, None)
+    ch.static_validation(req)
+
+
+def test_validate_context_pass_context_w3c_examples_v1():
+    # test for https://www.w3.org/2018/credentials/examples/v1
+    validate_context(W3C_EXAMPLE_V1_CONTEXT)
+
+
+def test_static_validation_fail_invalid_type():
+    operation = {
+        "meta": {
+            "type": "context",
+            "name": "TestContext",
+            "version": 1
+        },
+        "data": W3C_BASE_CONTEXT,
+        "type": "201"
+    }
+    req = Request("test", 1, operation, "sig",)
+    ch = ContextHandler(None, None)
+    with pytest.raises(LogicError):
+        ch.static_validation(req)
+
+
+def test_static_validation_fail_no_meta():
+    operation = {
+        "data": W3C_BASE_CONTEXT,
+        "type": "200"
+    }
+    req = Request("test", 1, operation, "sig", )
+    ch = ContextHandler(None, None)
+    with pytest.raises(KeyError) as e:
+        ch.static_validation(req)
+    assert 'meta' in str(e.value)
+
+
+def test_static_validation_fail_no_data():
+    operation = {
+        "meta": {
+            "type": CONTEXT_TYPE,
+            "name": "TestContext",
+            "version": 1
+        },
+        "type": "200"
+    }
+    req = Request("test", 1, operation, "sig", )
+    ch = ContextHandler(None, None)
+    with pytest.raises(KeyError) as e:
+        ch.static_validation(req)
+    assert "data" in str(e.value)
+
+
+def test_static_validation_fail_no_type(context_handler, context_request):
+    del context_request.operation['type']
+    with pytest.raises(LogicError):
+        context_handler.static_validation(context_request)
+
+
+def make_context_exist(context_request, context_handler):
+    identifier, req_id, operation = get_request_data(context_request)
+    context_name = get_write_context_name(context_request)
+    context_version = get_write_context_version(context_request)
+    path = ContextHandler.make_state_path_for_context(identifier, context_name, context_version)
+    context_handler.state.set(path, encode_state_value("value", "seqNo", "txnTime"))
+
+
+def test_context_dynamic_validation_failed_existing_context(context_request, context_handler):
+    make_context_exist(context_request, context_handler)
+    with pytest.raises(UnauthorizedClientRequest, match=str(AuthConstraintForbidden())):
+        context_handler.dynamic_validation(context_request)
+
+'''
```

Comment:
Please also add unit tests for `GetContextReqHandler`

Reply:
I'm having trouble finding equivalent unit tests for any of the other read request handlers (beyond the tests that call `get_result` in the `anon_creds` and `state_proof` tests, which have been implemented for CONTEXT). 
Could you point me to some?

Reply:
I believe the equivalent unit tests are the ones commented out below:
1) test that dynamic validation passes
2) test that dynamic validation fails
3) test for correct state update